http-client/Network/HTTP/Client/Request.hs

```diff
@@ -111,16 +113,30 @@ getUri req = URI
     , uriFragment = ""
     }
 
+applyAnyUriBasedAuth :: URI -> Request -> Request
+applyAnyUriBasedAuth uri =  applyBasicAuth (username authInfo) (password authInfo)
+ where
+   authInfo = uriUserInfo $ fromJust $ uriAuthority uri
```

Comment:
Probably just out of convenience to get to a passing test as quick as possible, played in part by my Haskell skill level. :)  I know this will fail with an error on `Nothing`.  Perhaps I should use `... $ fromMaybe "" $ ...` or is there a better way?  Wondering about something like `do uinfo <- uriAuthority uri; authInfo <- uriUserInfo uinfo; applyBasicAuth ...` (along with whatever signature changes necessary to deal with Maybe monad operators).